src/gui/gtk.c

```diff
@@ -773,6 +773,30 @@ static gboolean _scrollbar_changed(GtkWidget *widget,
   return TRUE;
 }
 
+gboolean _valid_window_placement(gint saved_x, gint saved_y, gint window_width, gint window_height)
+{
+  GdkDisplay *display = gdk_display_get_default();
+  gint n_monitors = gdk_display_get_n_monitors(display);
+
+  // check each monitor
+  for(gint i = 0; i < n_monitors; i++)
+  {
+    GdkMonitor *monitor = gdk_display_get_monitor(display, i);
+    GdkRectangle geometry;
+    gdk_monitor_get_geometry(monitor, &geometry);
+
+    // Check if window overlaps with this monitor
+    gboolean x_overlap = (saved_x < geometry.x + geometry.width) && (saved_x + window_width > geometry.x);
```

Comment:
Could you maybe be a little more picky and require that at least part of the header/top-left of the window is on the monitor? Otherwise you could still end up with just the bottom-right corner of the window being on the top-left of the monitor and that is a little hard to interact with (or even notice). I guess you'll have to allow for the border of the window to be outside of the monitor, since that is likely how full-screen is implemented.